distributed/client.py

```diff
@@ -643,6 +643,17 @@ def __init__(
             if address:
                 logger.info("Config value `scheduler-address` found: %s", address)
 
+        if address is not None and kwargs:
+            unused_arg = [
+                arg
+                for arg in kwargs
+                if arg not in ["silence_logs"] and kwargs.get(arg) is not None
+            ]
```

Comment:
I'm curious, why did you have to include these exceptions?  

My guess is that you're working around test failures somewhere in the test suite.  It's entirely possible though that those tests are wrong.

Reply:
Yes, this is just to get the current test suite to pass. Some tests like `test_client_repr_closed` include arguments that cause the test to fail:

```
c = yield Client(s.address, asynchronous=True, dashboard_address=None)
```

While `dashboard_address` doesn't have a value, kwargs is no longer None/False so it's going to raise an exception. [Here's the example in travis logs.](https://travis-ci.org/dask/distributed/jobs/593684024#L3253)

I think there are only 2 or 3 tests that would need to be changed. I can do that if that's the better than working around them.

Reply:
> I think there are only 2 or 3 tests that would need to be changed. I can do that if that's the better than working around them.

Yes, that would be ideal if you're willing.  Thank you for fixing more bugs :)